src/dirwatcher.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const fsPromises = require('fs/promises');
+const { EventEmitter } = require('events');
+
+class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+    this.directoryContents = {};
```

Comment:
Use es6 `Map` class instead of plain objects in case if you want to store key-value pairs.